ops/terraform/services/migrator/Jenkinsfile

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env groovy
+
+// entrypoint to migrator deployment, requires mapped arguments and an aws authentication closure
+// attempts to deploy and monitor and return `true` when the migrator signals a zero exit status
+boolean deployMigrator(Map args = [:], authFunction) {
+    amiId = args.amiId
+    bfdEnv = args.bfdEnv
+    heartbeatInterval = args.heartbeatInterval ?: 30
+    awsRegion = args.awsRegion ?: 'us-east-1'
+    gitBranchName = args.gitBranchName
+
+    // authenticate
+    authFunction()
+
+    // set sqsQueueName
+    sqsQueueName = "${bfdEnv}-db-migrator"
+
+    // precheck
+    if (canMigratorDeploymentProceed(sqsQueueName, awsRegion)) {
+        println "Proceeding to Migrator Deployment"
+    } else {
+        println "Halting Migrator Deployment. Check the SQS Queue ${sqsQueueName}."
+        return false
+    }
+
+    // plan/apply terraform
+    executeTerraform(amiId: amiId,
+                     bfdEnv: bfdEnv,
+                     gitBranchName: gitBranchName,
+                     heartbeatInterval: heartbeatInterval,
+                     createMigratorInstance: true)
+
+    // monitor migrator deployment
+    finalMigratorStatus = monitorMigrator(
+        sqsQueueName: sqsQueueName,
+        region: awsRegion,
+        heartbeatInterval: heartbeatInterval,
+        maxMessages: 10,
+        authFunction
+    )
+
+    // re-authenticate
+    authFunction()
+
+    // set return value for final disposition
+    if (finalMigratorStatus == '0') {
+        migratorDeployedSuccessfully = true
+        // Teardown when there is a healthy exit status
+        executeTerraform(amiId: amiId, bfdEnv: bfdEnv, gitBranchName: gitBranchName)
+    } else {
+        migratorDeployedSuccessfully = false
+    }
+
+    purgeQueue(sqsQueueName)
+
+    println "Migrator completed with exit status ${finalMigratorStatus}"
+    return migratorDeployedSuccessfully
+}
+
+// wraps the aws cli's receive-message subcommand to produce JSON objects
+// that represent the deployed migrator's state. Contains the following
+// fields mapped to string values: pid, start_time, stop_time, status, code
```

Comment:
can this comment be enhanced a bit to discuss the input parameters - maxMessages, the timeout, and the waitTime and how they affect the behavior of the function?